sdk/cosmos/sdk/src/main/java/com/azure/data/cosmos/sync/CosmosSyncClient.java

```diff
@@ -0,0 +1,230 @@
+        return mapDatabaseResponseAndBlock(asyncClientWrapper.createDatabase(id, throughput));
+    }
+
+    CosmosSyncDatabaseResponse mapDatabaseResponseAndBlock(Mono<CosmosDatabaseResponse> databaseMono)
+            throws CosmosClientException {
+        try {
+            return databaseMono
+                           .map(this::convertResponse)
+                           .block();
+        } catch (Exception ex) {
+            final Throwable throwable = Exceptions.unwrap(ex);
+            if (throwable instanceof CosmosClientException) {
+                throw (CosmosClientException) throwable;
+            } else {
+                throw ex;
+            }
+        }
+    }
+
+    /**
+     * Reads all databases.
+     *
+     * @param options {@link FeedOptions}the feed options.
+     * @return the iterator for feed response with the read databases.
+     */
+    public Iterator<FeedResponse<CosmosDatabaseProperties>> readAllDatabases(FeedOptions options) {
+        return asyncClientWrapper.readAllDatabases(options)
+                       .toIterable()
+                       .iterator();
+    }
+
+    /**
+     * Reads all databases.
+     *
+     * @return the iterator for feed response with the read databases.
+     */
+    public Iterator<FeedResponse<CosmosDatabaseProperties>> readAllDatabases() {
+        return asyncClientWrapper.readAllDatabases()
+                       .toIterable()
+                       .iterator();
+    }
+
+    /**
+     * Query a database
+     *
+     * @param query the query
+     * @param options {@link FeedOptions}the feed options.
+     * @return the iterator for feed response with the obtained databases.
+     */
+    public Iterator<FeedResponse<CosmosDatabaseProperties>> queryDatabases(String query, FeedOptions options) {
+        return asyncClientWrapper.queryDatabases(query, options)
+                       .toIterable()
+                       .iterator();
+    }
+
+    /**
+     * Query a database
+     *
+     * @param querySpec {@link SqlQuerySpec} the query spec
+     * @param options the query
+     * @return the iterator for feed response with the obtained databases.
+     */
+    public Iterator<FeedResponse<CosmosDatabaseProperties>> queryDatabases(SqlQuerySpec querySpec, FeedOptions options) {
+        return asyncClientWrapper.queryDatabases(querySpec, options)
+                       .toIterable()
+                       .iterator();
+    }
+
+    /**
+     * Gets the database client
+     *
+     * @param id the id of the database
+     * @return {@link CosmosSyncDatabase} the cosmos sync database
+     */
+    public CosmosSyncDatabase getDatabase(String id) {
+        return new CosmosSyncDatabase(id, this, asyncClientWrapper.getDatabase(id));
+    }
+
+    CosmosSyncDatabaseResponse convertResponse(CosmosDatabaseResponse response) {
+        return new CosmosSyncDatabaseResponse(response, this);
+    }
+
+    CosmosClient getAsyncClient() {
```

Comment:
Since this is a getter, we should rename this to asyncClient(). We are trying to follow the pattern x() instead of getX()

Reply:
used get since this is not public. Refactored it.